tests/integration/test_rootcheck/test_configuration/test_rootcheck.py

```diff
@@ -0,0 +1,253 @@
+
+components:
+    - rootcheck
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-analysisd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/capabilities/policy-monitoring/rootcheck
+    - https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-analysisd.html
+
+tags:
+    - rootcheck
+'''
+
+import json
+import os
+import time
+import pytest
+from pathlib import Path
+
+from wazuh_testing.constants.paths.logs import ALERTS_JSON_PATH
+from wazuh_testing.constants.paths.sockets import QUEUE_DB_PATH, WAZUH_DB_SOCKET_PATH
+from wazuh_testing.tools.db_administrator import DatabaseAdministrator
+from wazuh_testing.tools.simulators.agent_simulator import Sender, Injector, create_agents, connect
+from wazuh_testing.tools.socket_controller import SocketController
+from wazuh_testing.utils import configuration
+from wazuh_testing.utils.network import TCP
+from wazuh_testing.utils.services import control_service
+from wazuh_testing.utils.manage_agents import remove_agents
+
+
+from . import CONFIGS_PATH, TEST_CASES_PATH
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+injectors = []
+agents = []
+
+def retrieve_rootcheck_rows(agent_id):
+    db_connection = DatabaseAdministrator(os.path.join(QUEUE_DB_PATH, f'{agent_id}.db'))
+    rows = db_connection.select("pm_event")
+    db_connection.cursor.close()
+    db_connection.connection.close()
+    return rows
+
+def send_delete_table_request(agent_id):
+    controller = SocketController(WAZUH_DB_SOCKET_PATH)
+    controller.send(f'agent {agent_id} rootcheck delete', size=True)
+    response = controller.receive(size=True)
+    return response
+
+# Configuration and cases data.
+test_configs_path = Path(CONFIGS_PATH, 'config_template.yaml')
+test_cases_path = Path(TEST_CASES_PATH, 'cases_configuration.yaml')
+
+# Test configurations.
+test_configuration, test_metadata, test_cases_ids = configuration.get_test_cases_data(test_cases_path)
+test_configuration = configuration.load_configuration_template(test_configs_path, test_configuration, test_metadata)
+
+# Test daemons to restart.
+daemons_handler_configuration = {'all_daemons': True}
+
+
+@pytest.fixture(scope="function")
+def load_agents(request):
+
+    agents_number = request.getfixturevalue("test_metadata")["agents_number"]
+
+    for index in range(agents_number):
+        agent = create_agents(1, 'localhost')[0]
+        agent.modules['rootcheck']['status'] = 'enabled'
+        sender = Sender(manager_address='localhost', protocol=TCP, manager_port='1514')
+        injector = Injector(sender, agent)
+        injectors.append(injector)
+        injector.run()
+        agent.wait_status_active()
```

Comment:
This is mostly the **connect** (from agent_simulator) function but if I call it, then I got the warning 

```
/usr/local/lib/python3.10/dist-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-025rootcheck
  
  Traceback (most recent call last):
    File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
      self.run()
    File "/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/simulators/agent_simulator.py", line 1710, in run
      self.run_module(self.module)
    File "/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/simulators/agent_simulator.py", line 1690, in run_module
      self.sender.send_event(event)
    File "/usr/local/lib/python3.10/dist-packages/wazuh_testing/tools/simulators/agent_simulator.py", line 1513, in send_event
      self.socket.send(length + event)
  OSError: [Errno 9] Bad file descriptor
  ```